### ISSUE TITLE
.overlay with bokeh and .apply not displaying plot

### ISSUE THREAD

**Issue by DancingQuanta** [holoviz/holoviews#4217]
#### ALL software version info (this library, plus any other relevant software, e.g. bokeh, python, notebook, OS, browser, etc)
```
python: 3.7.6
holoviews: 1.13.0a22
hvplot: 0.5.3a1
bokeh: 1.4.0
jupyterlab: 1.4.0
notebook: 6.0.3
chrome: Version 79.0.3945.130 (Official Build) (64-bit)
```
#### Description of expected behavior and the observed behavior
I was trying to modify opts of a scatter plot based on a dimension (`direction` dimension with two values) with `.apply(fn)` then overlaying along the same or different dimension via `.overlay()`. With bokeh in jupyterlab and classic jupyter notebook the plot do not display at all or displays only a toolbar. Matplotlib worked fiine.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import numpy as np
import pandas as pd
import xarray as xr
import scipy.stats as stats

import holoviews as hv
from holoviews import opts
import hvplot.pandas

hv.extension('bokeh')

# Normal distribution parameters
variance = 1
sigma = np.sqrt(variance)
mu = xr.DataArray([-1, 1], coords={'direction': ['negative', 'positive']}, dims='direction')

# Independent variable
x = xr.DataArray(np.linspace(0, 1, len(idx)), dims='x', name='x')
x = (x * sigma * 2 - sigma) * 3

# Generate normal distribution
y = xr.apply_ufunc(stats.norm.pdf, x, mu, sigma)

# Add noise
noise = xr.DataArray(np.random.randn(len(V), len(I), len(direction), len(x)),
                 coords=[V, I, direction, x],
                 name='noise')
y = y + noise * 0.03
y.name = 'y'

def compute_marker(scatter):
    direction_markers = {
        'positive': 'circle',
        'negative': 'triangle'
    }
    sel_data = scatter.data
    direction = sel_data['direction'].iloc[0]
    return scatter.opts(marker=direction_markers[direction])

plot = (y.to_dataframe()
        .hvplot.scatter(x='x', y='y', groupby=['V', 'I', 'direction'], dynamic=False)
        .apply(compute_marker))

plot.overlay('direction') # No display in output
plot.overlay('I') # No display in output
plot.overlay('I').layout('direction') # Only displays a bokeh toolbar
```

#### Stack traceback and/or browser JavaScript console output

```
VM463:431 [bokeh] setting log level to: 'info'
VM463:445 Uncaught Error: invalid bbox {left: 52, top: 33, right: -6000, bottom: 255}
    at new t (<anonymous>:445:1219)
    at c (<anonymous>:558:8814)
    at <anonymous>:558:8882
    at t.foreach (<anonymous>:558:1342)
    at i._set_geometry (<anonymous>:558:8644)
    at i.set_geometry (<anonymous>:556:1673)
    at i.compute (<anonymous>:556:3234)
    at i.compute_layout (<anonymous>:611:3205)
    at i.build (<anonymous>:611:3033)
    at i.renderTo (<anonymous>:611:2929)
t @ VM463:445
c @ VM463:558
(anonymous) @ VM463:558
t.foreach @ VM463:558
i._set_geometry @ VM463:558
i.set_geometry @ VM463:556
i.compute @ VM463:556
i.compute_layout @ VM463:611
i.build @ VM463:611
i.renderTo @ VM463:611
v @ VM463:719
t.add_document_standalone @ VM463:719
t.embed_items_notebook @ VM463:726
embed_document @ VM466:56
(anonymous) @ VM466:59
(anonymous) @ VM466:75
push.HWBy.PanelLayout.attachWidget @ panellayout.js:213
push.HWBy.PanelLayout.insertWidget @ panellayout.js:116
_insertOutput @ widget.js:393
onModelChanged @ widget.js:217
invokeSlot @ index.js:476
emit @ index.js:433
push.qUp9.Signal.emit @ index.js:106
_onListChanged @ model.js:236
invokeSlot @ index.js:476
emit @ index.js:433
push.qUp9.Signal.emit @ index.js:106
push @ observablelist.js:135
_add @ model.js:212
add @ model.js:133
OutputArea._onIOPub @ widget.js:91
_handleIOPub @ future.js:204
async function (async)
_handleIOPub @ future.js:200
handleMsg @ future.js:174
_handleMessage @ default.js:1067
(anonymous) @ default.js:122
Promise.then (async)
DefaultKernel._onWSMessage @ default.js:119
Show 2 more frames
```

**Comment by philippjfr** [comment]
Your example is sadly not reproducible, various variables are not defined.

**Comment by philippjfr** [comment]
Somewhat hopeful this has been fixed in https://github.com/holoviz/holoviews/pull/4249.

**Comment by DancingQuanta** [comment]
Apologies for the incomplete example.
Here is the full example
```
import numpy as np
import pandas as pd
import xarray as xr
import scipy.stats as stats

import holoviews as hv
from holoviews import opts
import hvplot.pandas

hv.extension('bokeh')

# Normal distribution parameters
variance = 1
sigma = np.sqrt(variance)
mu = xr.DataArray([-1, 1], coords={'direction': ['negative', 'positive']}, dims='direction')

# Independent variable
x = xr.DataArray(np.linspace(0, 1, len(idx)), dims='x', name='x')
x = (x * sigma * 2 - sigma) * 3

# Generate normal distribution
y = xr.apply_ufunc(stats.norm.pdf, x, mu, sigma)

# Add noise
noise = xr.DataArray(np.random.randn(len(V), len(I), len(direction), len(x)),
                 coords=[V, I, direction, x],
                 name='noise')
y = y + noise * 0.03
y.name = 'y'

def compute_marker(scatter):
    direction_markers = {
        'positive': 'circle',
        'negative': 'triangle'
    }
    sel_data = scatter.data
    direction = sel_data['direction'].iloc[0]
    return scatter.opts(marker=direction_markers[direction])

plot = (y.to_dataframe()
        .hvplot.scatter(x='x', y='y', groupby=['V', 'I', 'direction'], dynamic=False)
        .apply(compute_marker))

plot.overlay('direction') # No display in output
plot.overlay('I') # No display in output
plot.overlay('I').layout('direction') # Only displays a bokeh toolbar
```

**Comment by philippjfr** [comment]
Still looks incomplete to me, where are `idx`, `V`, `I` and `direction` defined?

**Comment by DancingQuanta** [comment]
Sorry, thought I copied the correct code
```
import numpy as np
import pandas as pd
import xarray as xr
import scipy.stats as stats

import holoviews as hv
from holoviews import opts
import hvplot.pandas

hv.extension('bokeh')

# Dimensions
I = xr.DataArray(np.arange(10), dims='I', name='I')
direction = xr.DataArray(['negative', 'positive'], dims='direction', name='direction')
reps = xr.DataArray(np.arange(5), dims='reps', name='reps')
idx = np.arange(100)

# Normal distribution parameters
variance = 1
sigma = np.sqrt(variance)
mu = xr.DataArray([-1, 1], dims='direction')

# Independent variable
x = xr.DataArray(np.linspace(0, 1, 100), dims='x', name='x')
x = (x * sigma * 2 - sigma) * 4

# Generate normal distribution
y = xr.apply_ufunc(stats.norm.pdf, x, mu, sigma)

# Add noise
noise = xr.DataArray(np.random.randn(len(I), len(reps), *y.shape),
                     dims=['I', 'reps', *y.dims],
                     name='noise')

y = y + noise * 0.03
y.name = 'y'

# Assign coords
y = y.assign_coords(x=x, reps=reps, direction=direction, I=I)

def compute_marker(scatter):
    direction_markers = {
        'positive': 'circle',
        'negative': 'triangle'
    }
    sel_data = scatter.data
    direction = sel_data['direction'].iloc[0]
    return scatter.opts(marker=direction_markers[direction])

plot = (y.to_dataframe()
        .hvplot.scatter(x='x', y='y', groupby=['I', 'direction', 'reps'], dynamic=False)
        .apply(compute_marker))

plot.overlay('direction') # No display in output
plot.overlay('I') # No display in output
plot.overlay('I').layout('direction') # Only displays a bokeh toolbar
```

**Comment by philippjfr** [comment]
Great thanks, can confirm this was fixed by #4249.
